src/utils.ts

```diff
@@ -0,0 +1,15 @@
+export function withTimeout<T>(promise: Promise<T>, timeoutMs?: number | undefined): Promise<T> {
```

Comment:
To double-check semantics of the other sdks - in other sdks, it looks like if we cancel an operation due to a timeout, we expect the operation to halt (using [`increment_epoch`](https://docs.rs/wasmtime/latest/wasmtime/struct.Engine.html#method.increment_epoch)).

If I understand this correctly, this is kind of a tricky feature to implement in JS. We can support this for `runInWorker: true` plugins – we can kill the worker thread on timeout – but I don't think we have a cancelable promise otherwise. This would be user-visible if the plugin has access to host functions.

Reply:
Great points, while I was able to implement it in the background plugin, the foreground plugin doesn't seem to work:
https://github.com/extism/js-sdk/pull/73/commits/43e75aceff971c145b353c26a278e877780c0d39

If I can't find a solution, I might ignore timeout for now and create a separate issue for it